plugins/modules/vcsa_settings.py

```diff
@@ -424,20 +424,27 @@ def __proxy(self):
 
         current = self.api_networking.Proxy.list()
         for p in proxy:
-            if not p['enabled']:
-                continue
-
             c = current[p['protocol']]
-            if c.server != p['url'] or c.port != p['port'] or c.username != p['username'] or p['password'] is not None or c.enabled != p['enabled']:
+            if p['enabled']:
+                change_required = any([
+                    c.enabled != p['enabled'],
+                    c.server != p['url'],
+                    c.port != p['port'],
+                    getattr(c, 'username', None) != p.get('username', None),
+                    p.get('password', None) is not None
+                ])
+            else:
+                change_required = bool(c.enabled != p['enabled'])
```

Comment:
```suggestion
            change_required = any([
                bool(c.enabled != p['enabled']),
                c.enabled != p['enabled'],
                c.server != p['url'],
                c.port != p['port'],
                getattr(c, 'username', None) != p.get('username', None),
                p.get('password', None) is not None
            ])
```

I don't understand why you're doing a `if p['enabled']` / `else` here. Wouldn't it be simpler to do this all in one go? Or am I overlooking something? Is there some corner case that this approach wouldn't address?

And why is it `p.get('password', None) is not None` instead of `p.get('password', None) is not None` and not `getattr(c, 'password', None) != p.get('password', None),`

@p3ck What do you think? A second opinion on this might be helpful.

Reply:
Thanks for the explanation, makes sense to me.

> The API does say it returns a password, but I know for other endpoints the password value is encrypted or not returned at all. I suppose I should investigate a bit more to confirm if we can compare the two values or not

I could also investigate this. But give me a few days. I'm pretty sure I won't be able to this tomorrow, but maybe the day after.

If the API returns the password, I think it would be better to compare it with the one given in the task. Otherwise, we'll always get a change when when a password is given. That's not really idempotent. If the API doesn't return the password, we can't do anything about it. But if it does, we can improve idempotency there :-)

Reply:
I am in no rush and appreciate the help :)

Reply:
I'm afraid we won't be able to check the current password against the one defined in the task. I've tested this today, and the API doesn't return the current password.

I think they're just re-using the data structure for _setting_ the configuration. It contains `password`, but this is always empty / `None`. But they don't return the password, probably for security reasons. At least this is the result from my tests.

Reply:
OK. Seems consistent with other API endpoints at least. I will update the module docs to reflect this

Reply:
Added an option that other modules have which allow sthe user to control how the password is handled. The default behavior is consistent with the current behavior. The module will always try to update the password since it cant check the value